docs/docs/reference/other-new-features/open-classes.md

```diff
@@ -0,0 +1,79 @@
+---
+layout: doc-page
+title: "Open Classes"
+---
+
+An `open` modifier on a class signals that the class is planned for extensions. Example:
+```scala
+// File Writer.scala
+package p
+
+open class Writer[T] {
+
+  /** Sends to stdout, can be overridden */
+  def send(x: T) = println(x)
+
+  /** Send all arguments using `send` */
+  def sendAll(xs: T*) = xs.foreach(send)
+}
+
+// File EncryptedWriter.scala
+package p
+
+class EncryptedWriter[T: Encryptable] extends Writer[T] {
+  override def send(x: T) = super.send(encrypt(x))
+}
+```
+An open class typically comes with some documentation that describes
+the internal calling patterns between methods of the class as well as hooks that can be overridden. We call this the _extension contract_ of the class. It is different from the _external contract_ between a class and its users.
+
+Classes that are not open can still be extended, but only if at least one of two alternative conditions is met:
+
+ - The extending class is in the same source file as the extended class. In this case, the extension is usually an internal implementation matter.
+
+ - The language feature `adhocExtensions` is enabled for the extending class. This is typically enabled by an import statement in the source file of the extension:
+   ```scala
+   import scala.language.adhocExtensions
+   ```
+   Alternatively, the feature can be enabled by the command line option `-language:adhocExtensions`.
+   If the feature is not enabled, the compiler will issue a "feature" warning. For instance, if the `open` modifier on class `Writer` is dropped, compiling `EncryptedWriter` would produce a warning:
+   ```
+   -- Feature Warning: EncryptedWriter.scala:6:14 ----
+     |class EncryptedWriter[T: Encryptable] extends Writer[T]
+     |                                              ^
+     |Unless class Writer is declared 'open', its extension in a separate file should be enabled
+     |by adding the import clause 'import scala.language.adhocExtensions'
+     |or by setting the compiler option -language:adhocExtensions.
+   ```
+
+### Motivation
+
+When writing a class, there are three possible expectations of extensibility:
+
+- The class is intended to allow extensions. This means one should expect
```

Comment:
```suggestion
1. The class is intended to allow extensions. This means one should expect
```
It would make your following statement where you reference these by numbers a little clearer.